docs/LanguageGuide.md

```diff
@@ -75,6 +76,7 @@ and other static analysis. The first three of them are also available in dynamic
 (do <expr1> <expr2> ... <return-expression>) ;; Perform side-effecting functions, then return a value
 (if <expression> <true-branch> <false-branch>) ;; Branching
 (while <expression> <body>) ;; Loop until expression is false
+(match <expression> <case1> <expression1> <case2> <expression2> ...) ;; Pattern match the first expression in a exhaustive cases
```

Comment:
`match` only works on sumtypes for now.

Reply:
It would be better to just add a note or change de entire comment?

Reply:
Good question. How about `Pattern matches <expression> against a set of sumtype constructors`? I don’t know whether that is clear enough, though.

Reply:
There is plans to support pattern match on non sumtypes?

> I don’t know whether that is clear enough, though.

It's clear enought for me

Reply:
Eventually yes, but we want to sort out a bunch of other fundamental things first. Contributions are always welcome, though! (Though this is likely to be a little too hard to add for someone who doesn’t have a deep understanding of the Carp compiler)